vcd/config.go

```diff
@@ -90,6 +91,55 @@ func debugPrintf(format string, args ...interface{}) {
 	}
 }
 
+// This is a global MutexKV for all resources
+var vcdMutexKV = mutexkv.NewMutexKV()
+
+func (cli *VCDClient) lockvApp(d *schema.ResourceData) {
+	key := fmt.Sprintf("org:%s|vdc:%s|vapp:%s", cli.getOrgName(d), cli.getVdcName(d), d.Get("name").(string))
+	vcdMutexKV.Lock(key)
+}
+
+func (cli *VCDClient) unLockvApp(d *schema.ResourceData) {
+	key := fmt.Sprintf("org:%s|vdc:%s|vapp:%s", cli.getOrgName(d), cli.getVdcName(d), d.Get("name").(string))
+	vcdMutexKV.Unlock(key)
+}
+
+func (cli *VCDClient) lockParentvApp(d *schema.ResourceData) {
```

Comment:
What is the reason of having "Parent" in these functions? Why not just `lockVApp`, `lockEdgeGateway` and so on?

BTW, we use "VApp" in function names (like FindVAppByName), while "vApp" in comments and messages.

Reply:
Linas, if you look to content of functions you will see difference what resource names are used to construct key.

Regarding function names - reverting back, seems you forgot that you asked to change to current ones.

Reply:
I did not forget, I asked to use "vApp" "in comments and messages" where they are used as stand-alone words, in function names it's a different story.

Reply:
Thanks for addressing feedback and adding checks to the functions that the values would exist in the maps.